### ISSUE TITLE
Throw exceptions when `null` is used in the simple form of CASE clause

### ISSUE THREAD

**Issue by antonio2368** [memgraph/memgraph#319]
CASE doesn't work well with null. This works as expected and prints 'works':
```
WITH 2 AS name
RETURN CASE name
    WHEN 2 THEN 'works'
    WHEN null THEN "doesn't work"
    ELSE 'something went wrong'
END
```
If we swap the first case from 2 to 3. It should print 'something went wrong', but instead it prints "doesn't work":
```
WITH 2 AS name
RETURN CASE name
    WHEN 3 THEN 'works'
    WHEN null THEN "doesn't work"
    ELSE 'something went wrong'
END
```

The correct way to write the query above would be using the generic form:
```
WITH 2 AS name
RETURN CASE
  WHEN name is NULL THEN "doesn't work"
  WHEN name = 2 THEN "works"
  ELSE "something went wrong"
END
```

User should get an error to prefer to general form when he has `NULL` in the simple form.

**Comment by gitbuda** [comment]
Related to #432

**Comment by gitbuda** [comment]
Closed in #432
